src/nupic/algorithms/Connections.hpp

```diff
@@ -366,6 +376,11 @@ class Connections : public Serializable
   std::vector<Synapse>
   synapsesForPresynapticCell(CellIdx presynapticCell) const;
 
+  /**
+   * For use with time-series datasets.
+   */
+  void reset();
```

Comment:
nit, wrap the reset code in `if(timiseries_) {} else NTA_WARN << "TM::reset() needed only when timeseries param is ON."`